srcDoc/custom_types.rst

```diff
@@ -0,0 +1,578 @@
+        string<SIZE_TEST_STRING> test_string;
+    };//@Extensibility FINAL_EXTENSIBILITY
+
+    struct SeqTest {
+        sequence<long, SIZE_TEST_SEQ> test_seq;
+    };//@Extensibility FINAL_EXTENSIBILITY
+
+    struct Test {
+        long test_long;
+        TestEnum test_enum;
+        StringTest test_string;
+        SeqTest test_seq;
+    };//@Extensibility FINAL_EXTENSIBILITY
+
+These functions in the example are spreaded across several `idl` files to show the
+multiple files capability of the feature.
+
+These are the steps needed to use the above type in *RTI Perftest* for the
+C++ (Traditional) API:
+
+1. Copy all the `idl` files into the `srcIdl/customType/` folder.
+
+2. The following functions should be implemented (optionally) to properly
+   initialize and set the custom type structures.
+
+    - **initialize_custom_type_data**
+    This function is used to initialize your data.
+    Using this function, you will be able to allocate memory or set an immutable
+    field of the data.
+    The function takes one argument:
+
+        - A reference to custom type data.
+
+    .. code-block:: c
+
+        bool initialize_custom_type_data(RTI_CUSTOM_TYPE &data)
+        {
+            bool success = true;
+            if (!data.test_seq.test_seq.ensure_length(SIZE_TEST_SEQ, SIZE_TEST_SEQ)) {
+                success = false;
+            }
+            data.test_enum = ENUM1;
+            return success;
+        }
+
+    - **register_custom_type_data**
+    This function is used to set your data before being registered. It is only
+    required for key types. Set the key field of the data based on the key input.
+    There is a one-to-one mapping between an input key
+    and an instance.
+    The function takes two arguments:
+
+        - A reference to custom type data.
+        - A specific number unique for every key.
+
+    .. code-block:: c
+
+        void register_custom_type_data(RTI_CUSTOM_TYPE & data, unsigned long key)
+        {
+            data.test_long = key;
+        }
+
+    - **set_custom_type_data**
+    This function is used to set your data before it is sent.
+    It is called every time the data is sent.
+    You must set the custom type data before it is sent with the right
+    "key" value and the "targetDataLen".
+    The function takes three arguments:
+
+        - A reference to custom type data.
+        - A specific number unique for every key.
+        - The target size set by the command-line parameter ``-dataLen <bytes>``
+        minus the overhead of *RTI Perftest*. If applicable, you can use this
+        value to set the content of the data.
+
+    .. code-block:: c
+
+        bool set_custom_type_data(
+                RTI_CUSTOM_TYPE & data,
+                unsigned long key,
+                int targetDataLen)
+        {
+            bool success = true;
+            data.test_long = key;
+            if (sprintf(data.test_string.test_string, "Hello World! %lu", key) < 0) {
```

Comment:
I would switch this with the contents of initialize.  The string should have been set once.  There's really no reason to set the test_string, over and over again.  It's not a key...well, I guess it would be in the key test since the "Test" struct itself has no keys, and the TestDataKeyed_t struct declares the nested "Test" struct as keyed, and thus all members are part of the key.  But that's not the point.  The point is that the embedded string probably should only need to be initialized once (sprintf is pretty expensive to call, and the test isn't really supposed to be measuring the cost of setting the value of the data being sent).  

On the other hand, this would be the logical place to use the targetDataLen to set the length of the sequence member.